pkg/vm/compatibility.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package vm
+
+import (
+	"encoding/json"
+	"errors"
+	"sort"
+	"strconv"
+
+	"github.com/ava-labs/avalanche-cli/pkg/application"
+	"github.com/ava-labs/avalanche-cli/pkg/binutils"
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"golang.org/x/mod/semver"
+)
+
+var ErrNoAvagoVersion = errors.New("unable to find a compatible avalanchego version")
+
+func GetRPCProtocolVersion(app *application.Avalanche, vmType models.VMType, vmVersion string) (int, error) {
+	var url string
+
+	switch vmType {
+	case models.SubnetEvm:
+		url = constants.SubnetEVMRPCCompatibilityURL
+	case models.SpacesVM:
+		url = constants.SpacesVMRPCCompatibilityURL
+	default:
+		return 0, errors.New("unknown VM type")
+	}
+
+	compatibilityBytes, err := app.Downloader.Download(url)
+	if err != nil {
+		return 0, err
+	}
+
+	var parsedCompat models.VMCompatibility
+	err = json.Unmarshal(compatibilityBytes, &parsedCompat)
+	if err != nil {
+		return 0, err
+	}
+
+	version, ok := parsedCompat.RPCChainVMProtocolVersion[vmVersion]
+	if !ok {
+		return 0, errors.New("no RPC version found")
+	}
+
+	return version, nil
+}
+
+func GetLatestAvalancheGoByProtocolVersion(app *application.Avalanche, rpcVersion int) (string, error) {
+	compatibilityBytes, err := app.Downloader.Download(constants.AvalancheGoCompatibilityURL)
+	if err != nil {
+		return "", err
+	}
+
+	var parsedCompat models.AvagoCompatiblity
+	err = json.Unmarshal(compatibilityBytes, &parsedCompat)
+	if err != nil {
+		return "", err
+	}
+
+	eligibleVersions, ok := parsedCompat[strconv.Itoa(rpcVersion)]
```

Comment:
Are these not semver versions? If they are, semver has inbuilt sorting if I remember correctly. Otherwise the sorting can actually get tricky.

Reply:
These are not actually semvers, they are just ints. Ex: https://github.com/ava-labs/subnet-evm/blob/master/compatibility.json